### PR TITLE
Support validating remote and external manifest stores

### PR DESCRIPTION
**Pull request by mauricefisher64**
## Changes in this pull request
The PR adds the ability to validate external .c2pa files and fetching and validating remote manifest stores.  Includes
small amount of code cleanup.

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [x] All applicable changes have been documented.
- [x] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/Cargo.toml [hunk 1]
```diff
@@ -25,6 +25,7 @@ file_io = ["openssl"]
 serialize_thumbnails = []
 xmp_write = ["xmp_toolkit"]
 no_interleaved_io = ["file_io"]
+fetch_remote_manifests = ["file_io"]
```
Comment (gpeacock) [inline sdk/Cargo.toml:28]: Ok, but, as we said, let's move away from adding features

File: sdk/src/claim.rs [hunk 2]
```diff
@@ -236,9 +236,10 @@ pub enum AssertionStoreJsonFormat {
 /// Remote manifest options. Use 'set_remote_manifest' to generate external manifests.
 #[derive(Clone, Debug, PartialEq)]
 pub enum RemoteManifest {
-    NoRemote,       // No external manifest (default)
+    NoRemote,                // No external manifest (default)
```
Comment (gpeacock) [inline sdk/src/claim.rs:239]: any reason for the extra space?

File: sdk/src/store.rs [hunk 3]
```diff
@@ -1734,69 +1761,67 @@ impl Store {
...
 
-        let cai_loader = get_cailoader_handler(asset_type).ok_or(Error::UnsupportedType)?;
+    // fetch remote manifest if possible
+    #[cfg(not(target_arch = "wasm32"))]
+    fn fetch_remote_manifest(url: &str) -> Result<Vec<u8>> {
+        use conv::ValueFrom;
+        use std::io::Read;
+
+        //const MANIFEST_CONTENT_TYPE: &str = "application/x-c2pa-manifest-store"; // todo verify once these are served
+        const DEFAULT_MANIFEST_RESPONSE_SIZE: usize = 10 * 1024 * 1024; // 10 MB
```
Comment (gpeacock) [inline sdk/src/store.rs:1784]: I don't know what a reasonable limit would be for this, but I guess we will find out.

File: sdk/src/store.rs [hunk 4]
```diff
@@ -1734,69 +1761,67 @@ impl Store {
...
-            self,
-            xmp_opt,
-            &ClaimAssetData::ByteData(buf),
-            validation_log,
-        )?;
+                    response
+                        .into_reader()
+                        .take(len64)
+                        .read_to_end(&mut response_bytes)
+                        .map_err(|_err| Error::CoseTimeStampGeneration)?;
```
Comment (gpeacock) [inline sdk/src/store.rs:1804]: Why CoseTimeStampGeneration error here?

**Review by gpeacock — CHANGES_REQUESTED** [review]

File: sdk/src/claim.rs (hunk 2 above)
Comment (mauricefisher64) [reply sdk/src/claim.rs:239]: every time I remove it cargo fmt puts it back

File: sdk/src/store.rs (hunk 4 above)
Comment (mauricefisher64) [reply sdk/src/store.rs:1804]: It should not be that.  Auto-fill got me.

File: sdk/Cargo.toml (hunk 1 above)
Comment (mauricefisher64) [reply sdk/Cargo.toml:28]: Yes, we do not have a way to control this yet.

File: sdk/src/store.rs (hunk 3 above)
Comment (mauricefisher64) [reply sdk/src/store.rs:1784]: I am going to expose this as an option when I add the config stuff.

**Review by gpeacock — APPROVED** [review]

**Pull request merged**